### PR TITLE
Add syntax-fm extension

### PR DESCRIPTION
**Pull request by msmps**
## Description

The aim of this extension is to have [one of my favourite podcasts](https://x.com/msmps_/status/1730711771722232245) syntax.fm at your fingertips! ✨

It currently supports searching for episodes, opening directly on the syntax.fm website, opening on YouTube (_if a video is present_) and finally opening in the Apple Music application.

I plan on iterating on this, adding background checks to check for new episodes, and adding support to open directly in Spotify, but first I need to open an issue or contribute back to syntax to include the Spotify information for each track.

I think my only outstanding question is how to best cache the data from the `useFetch` hook and/or if we should consider pagination at this stage. The endpoint I'm using doesn't currently support pagination.

## Screencast

https://github.com/raycast/extensions/assets/7691252/d4c9c1b0-9158-4459-a7e7-d206ad26d8d8

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by msmps** [comment]
Hey @wesbos @stolinski 👋  I read through the [Syntax Terms of Service](https://syntax.fm/pages/terms-of-service), and nothing stood out to me. I just want to make sure you're both happy with this and the consumption of the content.json endpoint.

**Comment by stolinski** [comment]
Sick! Thank you @msmps

File: extensions/syntax-fm/src/search.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                      <Action.OpenInBrowser
+                        icon={Icon.Video}
+                        url={item.youtube_url}
+                        title="Watch on YouTube"
+                        shortcut={{ modifiers: ["cmd", "shift"], key: "v" }}
+                      />
+                    )}
+                    <Action.Push
+                      icon={Icon.Document}
+                      title="View Show Notes"
```
Comment (pernielsentikaer) [inline extensions/syntax-fm/src/search.tsx:39]: View or Show? 🙂

File: extensions/syntax-fm/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  ],
+  "keywords": [
+    "podcast",
+    "syntax"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "search",
+      "title": "Search Episodes",
```
Comment (pernielsentikaer) [inline extensions/syntax-fm/package.json:20]:
```suggestion
      "title": "Search Episodes",
      "subtitle": "Syntax.fm",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

The data is already cached when you're using useFetch from @raycast/utils, it will show the cached data while getting fresh data from the endpoint so it should be fine 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/syntax-fm/src/search.tsx (hunk 1 above)
Comment (msmps) [reply extensions/syntax-fm/src/search.tsx:39]: Thank you for the review @pernielsentikaer

That's a tough one; the notes are actually called "Show Notes" and that's what listeners refer to them as, which is why I went for this wording to "View the Show Notes".

image

Comment (pernielsentikaer) [reply extensions/syntax-fm/src/search.tsx:39]: That makes a lot more sense now 🙂

**Comment by wesbos** [comment]
DAMN this is SO cool! Thanks a ton for doing this. 

We have a new logo, I'm gonna PR it right now

**Comment by pernielsentikaer** [comment]
do-it-gif

**Comment by wesbos** [comment]
cloning the repo. Give me 3 days

**Pull request merged**